### PR TITLE
Fix rendering voice message if the waveform data is corrupted

### LINKED ISSUE #3983: Voice Message - Cannot render voice message if the waveform data is corrupted
### Steps to reproduce

This issue is created for a hotfix. The main issue is described in #3975.

### What did you expect?
To be able to playback voice messages in the timeline.

### What happened?
App cannot render the voice message item at all.

### Your phone model

_No response_

### Operating system version

_No response_

### Application version and app store

_No response_

### Homeserver

_No response_

### Have you submitted a rageshake?

Yes

### PR DESCRIPTION
**Pull request by onurays**
Fixes #3983

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/content/ContentAttachmentData.kt [hunk 1]
```diff
@@ -36,7 +36,7 @@ data class ContentAttachmentData(
         val queryUri: Uri,
         val mimeType: String?,
         val type: Type,
-        val waveform: List<Int>? = null
+        val waveform: List<Int?>? = null
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/content/ContentAttachmentData.kt:39]: This is only used locally, so I do not understand why you are changing the type here too.

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/model/message/AudioWaveformInfo.kt [hunk 2]
```diff
@@ -32,5 +32,5 @@ data class AudioWaveformInfo(
          * List of integers between zero and 1024, inclusive.
          */
         @Json(name = "waveform")
-        val waveform: List<Int>? = null
+        val waveform: List<Int?>? = null
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/model/message/AudioWaveformInfo.kt:35]: This change is OK

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/send/WaveFormSanitizer.kt [hunk 3]
```diff
@@ -33,7 +33,7 @@ internal class WaveFormSanitizer @Inject constructor() {
      * The array should have no less than 30 elements and no more than 120.
      * List of integers between zero and 1024, inclusive.
      */
-    fun sanitize(waveForm: List<Int>?): List<Int>? {
+    fun sanitize(waveForm: List<Int?>?): List<Int>? {
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/send/WaveFormSanitizer.kt:36]: This is use when we send a voice message, so I think this is not necessary to change anything in this file

**Review by bmarty — COMMENTED** [review]
A few remarks, maybe it's OK to keep it like that

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/content/ContentAttachmentData.kt (hunk 1 above)
Comment (onurays) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/content/ContentAttachmentData.kt:39]: You are right. Done.

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/send/WaveFormSanitizer.kt (hunk 3 above)
Comment (onurays) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/send/WaveFormSanitizer.kt:36]: Done.

**Review by bmarty — APPROVED** [review]
Thanks for the update

**Review by bmarty — APPROVED** [review]
Fix tested OK on my phone

**Pull request merged**